SevenRedGame/Program.cs

```
@@ -1,43 +1,84 @@
 
 namespace _7RedGame
 {
     class Program
     {
-        private static Game game = new Game(); 
+        private static Game game;
 
         static void Main(string[] args)
         {
-            GenerateCardCombo(1);
-            GenerateCardCombo(2);
-            var winningCard = game.GetWinningCombination();
-            Console.WriteLine($"{(winningCard.Value == 1 ? "First" : "Second") } combo won!");
-            Console.WriteLine(winningCard.Key.ToString());
+            try
+            {
+                Console.WriteLine("Enter card combinations count");
+                int combinationsCount;
+                while (!int.TryParse(Console.ReadLine(), out combinationsCount) || combinationsCount < 2)
+                {
+                    Console.WriteLine("Enter card combinations count");
+                }
+                game = new Game(combinationsCount);
+                GenerateCombinations(combinationsCount);
+                var winningCombo = game.GetWinningCombination();
+                Console.WriteLine($"{  winningCombo.Name } combo won!");
+                Console.WriteLine(game.GetHighestCardInCombination(winningCombo).ToString());
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.Message);
+            }
+        }
+
+        private static void GenerateCombinations(int count)
+        {
+            for (int i = 0; i < count; i++)
+            {
+                Console.WriteLine("Enter combination name");
+                var name = Console.ReadLine();
+                GenerateCardCombo(name);
+            }
         }
 
-        private static List<Card> GenerateCardCombo(int combinationNumber)
+        private static void GenerateCardCombo(string name)
         {
             Console.WriteLine("Enter combination count");
-            if(int.TryParse(Console.ReadLine(), out int comboCount))
+            if (int.TryParse(Console.ReadLine(), out int comboCount))
```

Instead of a large nesting, you can return the error here and then continue your flow. It will be better then return an error on  79 line